src/utilities/temperature.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2020
+ */
+
+/**
+ * Class representing a Temperature.
+ * Can have a different representation depending on it's options for
+ * easy converting between celsius and fahrenheit.
+ *
+ * @params value   {Number} The temperature
+ * @params options {Object} The temperatures options. Shape below.
+ *
+ * Options shape:
+ * {
+ *  metric: 'c' or 'f' - the current temperature metric. [Default: 'c']
+ *  precision: int - the number of decimal place precision. [Default: 1]
+ *  formatWithSymbol: bool - indicator if the metric symbol+symbol should be used.[Default: true]
+ *  pattern: A pattern for formatting. # is replaced with temp, * with the symbol. [Default: "#*"]
+ *  negativePattern: As above, for negative temperatures. [Default: "-#*"]
+ *  symbol: Symbol for formatting- shown between the temp and metric symbol [Default: '\u00B0']
+ * }
+ */
+class Temperature {
+  constructor(value = 0, options = {}) {
+    this.options = { ...DEFAULT_OPTIONS, ...options };
+    this.value = value;
+  }
+
+  temperature(asMetric) {
+    const { options, value } = this;
+    const { precision, metric } = options;
+
+    const metricToUse = asMetric || metric;
+
+    if (metricToUse === metric) return Number(value).toFixed(precision);
+    if (metricToUse === CELSIUS) return Number(celsiusFromFahrenheit(value)).toFixed(precision);
+
+    return Number(celsiusToFahrenheit(value)).toFixed(precision);
+  }
+
+  setMetric(newMetric) {
+    if (!(newMetric === CELSIUS || newMetric === FAHRENHEIT)) return this;
+    const { metric } = this.options;
+
+    if (newMetric !== metric) {
+      if (newMetric === CELSIUS) this.value = celsiusFromFahrenheit(this.value);
+      else this.value = celsiusToFahrenheit(this.value);
+    }
+
+    this.options.metric = newMetric;
+
+    return this;
+  }
+
+  format(asMetric) {
+    const { pattern, metric, symbol, negativePattern, formatWithSymbol } = this.options;
+
+    const formatMetric = asMetric || metric;
+    const temperatureValue = this.temperature(formatMetric);
+    const patternToReplace = temperatureValue >= 0 ? pattern : negativePattern;
+
+    if (!Number.isFinite(Number(temperatureValue))) return 'N/A';
```

Comment:
this might be overkill: the `N/A` could be either part of the options or set as a const at the top of file.

Reply:
nice! Added as an `invalidPattern` option.. maybe a bit strange? But gives more flexibility if ever for some reason needed - 691bf4f